java/data/worldgen/mod.mcdoc

```diff
@@ -43,30 +43,44 @@ type IntProvider<T> = (
 	T |
 	struct {
 		type: #[id="int_provider_type"] string,
-		#[until="1.20.5"]
-		value: minecraft:int_provider[[type]]<T>,
-		#[since="1.20.5"]
 		...minecraft:int_provider[[type]]<T>,
 	} |
 )
 
 type ConstantIntProvider<T> = struct {
 	value: T,
 }
-dispatch minecraft:int_provider[constant] to ConstantIntProvider
+
+dispatch minecraft:int_provider[constant] to (
+	#[until="1.20.5"] struct {
+		value: ConstantIntProvider
+	} |
+	ConstantIntProvider |
+)
 
 type UniformIntProvider<T> = struct {
 	min_inclusive: T,
 	max_inclusive: T,
 }
-dispatch minecraft:int_provider[uniform,biased_to_bottom] to UniformIntProvider
+
+dispatch minecraft:int_provider[uniform,biased_to_bottom]<T> to (
```

Comment:
Interesting, I didn't know this was valid syntax, but it seems like it is :D

Now that I know this, it might be possible to do this: (although what you have right now might be less confusing, and thus preferred)

```rs
dispatch minecraft:int_provider[uniform,biased_to_bottom]<T> to IntProviderSettings<UniformIntProvider, T>

type IntProviderSetting<P, T> = (
	#[until="1.20.5"] struct {
		value: P<T>
	} |
	#[since="1.20.5"] P<T> |
)
```

Reply:
Oh yeah, that would have probably been an option. But yeah, lets keep it like it is now - I think this is much easier to read.